bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/internal/DPIUtil.java

```diff
@@ -347,6 +347,18 @@ public static int getZoomForAutoscaleProperty (int nativeDeviceZoom) {
 	return getZoomForAutoscaleProperty(nativeDeviceZoom, autoScaleValue);
 }
 
+public static int getNativeZoomForAutoscaleProperty (int nativeDeviceZoom) {
+	int zoom = 0;
+	try {
+		// Native zoom is set to the autoscaled zoom if autoscale value uses an integer value 
+		Integer.parseInt (autoScaleValue);
```

Comment:
I am not sure if this is a good approach of dealing with the different cases of font zoom calculation:
1. The exception-driven logic is not that easy to understand.
2. The performance of exception-driven logic is bad and this will be called quite often with a thrown exception being the standard case

Wouldn't it make sense when changing the `autoScaleValue` to store a `fontZoomCalulator` that, depending on whether the `autoScaleValue` to be set is an integer, either gets the `getZoomForAutoscaleProperty` call assigned or just returns the given zoom. That would avoid that this logic to identify how to do the calculation is called everytime a zoom is to be convered.